src/main/java/org/carlmontrobotics/subsystems/Arm.java

```diff
@@ -1,48 +1,259 @@
+        armMotorMaster.setIdleMode(IdleMode.kBrake);
+        armMotorFollower.setInverted(MOTOR_INVERTED);
+        armMotorFollower.setIdleMode(IdleMode.kBrake);
+        
+        armEncoder.setPositionConversionFactor(ROTATION_TO_RAD);
+        armEncoder.setVelocityConversionFactor(ROTATION_TO_RAD);
+        armEncoder.setInverted(ENCODER_INVERTED);
+
+        armMotorFollower.follow(armMotorMaster);
+        armPID1.setP(kP);
+        armPID1.setI(kI);
+        armPID1.setD(kD);
+        armPID2.setP(kP);
+        armPID2.setI(kI);
+        armPID2.setD(kD);
+     
+        armEncoder.setZeroOffset(ENCODER_OFFSET_RAD);
+      
+        //armPID.setTolerance(posToleranceRad, velToleranceRadPSec);
+
+        armPID1.setFeedbackDevice(armMotorMaster.getAbsoluteEncoder(SparkAbsoluteEncoder.Type.kDutyCycle));
+        armPID1.setPositionPIDWrappingEnabled(true);
+        armPID1.setPositionPIDWrappingMinInput(ARM_LOWER_LIMIT_RAD);
+        armPID1.setPositionPIDWrappingMaxInput(ARM_UPPER_LIMIT_RAD);
+        //two PIDs?
+        armPID2.setFeedbackDevice(armMotorFollower.getEncoder());
+        armPID2.setPositionPIDWrappingEnabled(true);
+        armPID2.setPositionPIDWrappingMinInput(ARM_LOWER_LIMIT_RAD);
+        armPID2.setPositionPIDWrappingMaxInput(ARM_UPPER_LIMIT_RAD);
+
+        SmartDashboard.putData("Arm", this);
+
+        
+        
+        goalState = getCurrentArmState();
+
+        setArmTarget(goalState.position);
+    }
+
+    @Override
+    public void periodic() {
+
+        if(DriverStation.isDisabled()) resetGoal();
+
+        //ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD = SmartDashboard.getNumber("ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD", ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD);
+        // armConstraints = new TrapezoidProfile.Constraints(MAX_FF_VEL , MAX_FF_ACCEL );
+        
+
+        //smart dahsboard stuff
+        //SmartDashboard.putBoolean("ArmPIDAtSetpoint", armPID1.atSetpoint());
+        SmartDashboard.putBoolean("ArmProfileFinished", armProfile.isFinished(armProfileTimer.get()));
+        //posToleranceRad = SmartDashboard.getNumber("Arm Tolerance Pos", posToleranceRad);
+        //velToleranceRadPSec= SmartDashboard.getNumber("Arm Tolerance Vel", velToleranceRadPSec);
+
+       // SmartDashboard.putNumber("MaxHoldingTorque", maxHoldingTorqueNM());
+        //SmartDashboard.putNumber("V_PER_NM", getV_PER_NM());
+       // SmartDashboard.putNumber("COMDistance", getCoM().getNorm());
+        SmartDashboard.putNumber("InternalArmVelocity", armEncoder.getVelocity());
+        //SmartDashboard.putNumber("Arm Current", armMotor.getOutputCurrent());
+
+       // SmartDashboard.putNumber("ArmPos", getArmPos());
+
+
+
+        autoCancelArmCommand();
+    }
+
+    public TrapezoidProfile.State calculateCustomSetPoint(double goalSeconds, TrapezoidProfile.State currentPoint, TrapezoidProfile.State goalState) {
+        return armProfile.calculate(goalSeconds, currentPoint, goalState);
+        
+      }
```

Comment:
what is this for? I recommend using `setTargetPosition()` instead of using this method.